test/generation.test.js

```diff
@@ -16,8 +18,51 @@ describe('Generation tests', () => {
         httpVersion: '2',
     });
 
+    test('Generates unordered headers', () => {
+        const result = headerGenerator.getRawHeadersAndOrder();
+
+        expect(result).toBeTruthy();
+        expect(result.generatedSample).toBeTruthy();
+        expect(Array.isArray(result.order)).toBe(true);
+    });
+
     test('Generates headers', () => {
-        expect(headerGenerator.getHeaders()).toBeTruthy();
+        const headers = headerGenerator.getHeaders();
+
+        let userAgent;
+        for (const [header, value] of Object.entries(headers)) {
+            if (header.toLowerCase() === 'user-agent') {
+                userAgent = value;
+                break;
+            }
+        }
+
+        let browser;
+        if (userAgent.includes('Firefox')) {
+            browser = 'firefox';
+        } else if (userAgent.includes('Chrome')) {
+            browser = 'chrome';
+        } else {
+            browser = 'safari';
+        }
+
+        expect(typeof headers).toBe('object');
+
+        const order = headersOrder[browser];
+
+        let index = -1;
+        for (const header of Object.keys(headers)) {
+            const newIndex = order.indexOf(header);
+
+            const log = `${userAgent}\n${browser}\n${inspect(order)}\n${inspect(headers)}`;
+            if (newIndex === -1) {
+                throw new Error(`Missing entry in order array\n${log}`);
+            } else if (newIndex < index) {
```

Comment:
If the index of the header is earlier than the remembered index then throw an error. Example:

```
order:
 - Connection = 0
 - connection = 1
 - User-Agent = 2
 - user-agent = 3

headers:
User-Agent: ...
Connection: ...
```

so this looks up the index of the first header (`User-Agent`) which is `2`. Then it looks up the index for the next header. In this case it's `Connection`, which is `0`. `0 < 2` so it would throw.